cloud/deployment/workerqueue/workerqueue.go

```diff
@@ -35,264 +35,329 @@ var (
+			MinWorkerCount:    wQueueMin,         // use the value from the user input
+			MaxWorkerCount:    wQueueMax,         // use the value from the user input
+			WorkerConcurrency: wQueueConcurrency, // use the value from the user input
+		}
+		// create hybridListToCreate
+		for i := range *requestedDeployment.WorkerQueues {
+			queues := *requestedDeployment.WorkerQueues
+			existingHybridQueueRequest := astroplatformcore.HybridWorkerQueueRequest{
+				Name:              queues[i].Name,
+				Id:                &queues[i].Id,
+				IsDefault:         queues[i].IsDefault,
+				MaxWorkerCount:    queues[i].MaxWorkerCount,
+				MinWorkerCount:    queues[i].MinWorkerCount,
+				WorkerConcurrency: queues[i].WorkerConcurrency,
+				NodePoolId:        *queues[i].NodePoolId,
+			}
+			hybridListToCreate = append(hybridListToCreate, existingHybridQueueRequest)
+		}
+		if name == "" {
+			queueToCreateOrUpdateHybrid.Name, err = getQueueName(name, action, &requestedDeployment, out)
+			if err != nil {
+				return err
+			}
+			name = queueToCreateOrUpdateHybrid.Name
+		}
+		queueToCreateOrUpdateHybrid = SetWorkerQueueValuesHybrid(wQueueMin, wQueueMax, wQueueConcurrency, queueToCreateOrUpdateHybrid, defaultOptions)
 	}
-	switch requestedDeployment.DeploymentSpec.Executor {
-	case deployment.CeleryExecutor:
-		// get defaults for min-count, max-count and concurrency from API
-		defaultOptions, err = GetWorkerQueueDefaultOptions(client)
+	switch *requestedDeployment.Executor {
+	case astroplatformcore.DeploymentExecutorCELERY:
+		deploymentOptions, err := deployment.GetPlatformDeploymentOptions("", astroplatformcore.GetDeploymentOptionsParams{}, platformCoreClient)
 		if err != nil {
 			return err
 		}
 
-		queueToCreateOrUpdate = SetWorkerQueueValues(wQueueMin, wQueueMax, wQueueConcurrency, queueToCreateOrUpdate, defaultOptions)
-		if deployment.IsDeploymentHosted(requestedDeployment.Type) || deployment.IsDeploymentDedicated(requestedDeployment.Type) {
-			err = IsHostedCeleryWorkerQueueInputValid(queueToCreateOrUpdate, defaultOptions, &astroMachine)
+		defaultOptions = deploymentOptions.WorkerQueues
+
+		if deployment.IsDeploymentStandard(*requestedDeployment.Type) || deployment.IsDeploymentDedicated(*requestedDeployment.Type) {
+			err = IsHostedCeleryWorkerQueueInputValid(queueToCreateOrUpdate, defaultOptions, &workerMachine)
 			if err != nil {
 				return err
 			}
 		} else {
-			err = IsCeleryWorkerQueueInputValid(queueToCreateOrUpdate, defaultOptions)
+			err = IsCeleryWorkerQueueInputValid(queueToCreateOrUpdateHybrid, defaultOptions)
 			if err != nil {
 				return err
 			}
 		}
-	case deployment.KubeExecutor:
-		err = IsKubernetesWorkerQueueInputValid(queueToCreateOrUpdate)
+	case astroplatformcore.DeploymentExecutorKUBERNETES:
+		if deployment.IsDeploymentStandard(*requestedDeployment.Type) || deployment.IsDeploymentDedicated(*requestedDeployment.Type) {
+			return errNoUseWorkerQueues
+		}
+		queueToCreateOrUpdateHybrid.MinWorkerCount = -1
```

Comment:
can you write a comment as to why we do this?

Reply:
added a comment